freeRTOS_EDF-Mustafa/FreeRTOS/Source/tasks.c

```diff
@@ -2945,30 +2962,35 @@ BaseType_t xSwitchRequired = pdFALSE;
 						only be performed if the unblocked task has a
 						priority that is equal to or higher than the
 						currently executing task. */
-						#if ( configUSE_EDF_SCHEDULER == 0 )
-						{
-						if( pxTCB->uxPriority >= pxCurrentTCB->uxPriority )
-						{
-							xSwitchRequired = pdTRUE;
-						}
-						else
+						/*...NEW... */
+						/* as soon as a new task is available in the EDF ready list,
+						a context switching should take place. Modify preemption way as any task
+						with a sooner deadline must preempt a task with a larger deadline
+						instead of a priority
+						*/
+						#if (configUSE_EDF_SCHEDULER == 1)
 						{
-							mtCOVERAGE_TEST_MARKER();
-						}
+							if( pxTCB->xTaskPeriod <= pxCurrentTCB->xTaskPeriod )
```

Comment:
Here you have to make a condition that the task with the sooner deadline must preempt the task with a larger deadline

Note: -Deadline is stored in a variable named "xItemValue"